cpdb/cpdb.c

```diff
@@ -258,8 +258,9 @@ char *cpdbGetSysConfDir()
         free(env_xcd);
     }
         
-    config_dir = "/etc/cpdb";
-    if (access(config_dir, R_OK) == 0 || mkdir(config_dir, CPDB_SYSCONFDIR_PERM) == 0)
+    const char* static_path = "/etc/cpdb";
+    if (access(static_path, R_OK) == 0 || mkdir(static_path, CPDB_SYSCONFDIR_PERM) == 0)
+        config_dir = g_strdup(static_path);
         return config_dir;
```

Comment:
Since there are no braces, only the `config_dir = g_strdup(static_path);` is part of the `if` block, the return statement is not, which means that a previously freed `config_dir` would be returned if the if check fails.

Need to either just `return g_strdup(static_path);` or add `{ ... }`.